nncf/common/tensor_statistics/statistic_point.py

```diff
@@ -71,6 +71,26 @@ def add_statistic_point(self, statistic_point: StatisticPoint) -> None:
 
             self.data[target_node_name].append(statistic_point)
 
+    def remove_statistic_points(self, algorithm_name: str) -> None:
+        """
+        Method to remove statistic point associated with a given algorithm
+        from statistic point container.
+
+        :param algorithm: Algorithm name
+        """
+        data_copy = dict(self.data)
```

Comment:
Why it is necessary to create copy here?

Reply:
> Why it is necessary to create copy here?

To avoid iterating through the **original dictionary** while modifying it at the same time